plugins/inputs/internet_speed/internet_speed.go

```diff
@@ -83,6 +83,7 @@ func (is *InternetSpeed) Gather(acc telegraf.Accumulator) error {
 	fields["download"] = s.DLSpeed
 	fields["upload"] = s.ULSpeed
 	fields["latency"] = timeDurationMillisecondToFloat64(s.Latency)
+	fields["provider"] = s.Sponsor
```

Comment:
traditionally strings are normally marked as tags and not fields. Was there a reason you went with a field?

Reply:
> traditionally strings are normally marked as tags and not fields. Was there a reason you went with a field?
> 
> Also, so I am clear, the "sponsor" is the server you are connected to? Or is it your own internet provider? The terminology isn't clear.

Sure!

Sponsor is Ookla's terminology for the ISP(rovider) hosting the speedtest instance (or the one the client is currently testing with). I chose a field because the sponsor/provider is randomized and changes (That's what's helpful to track/store/report).

Reply:
Thanks! Let's call this [source](https://github.com/influxdata/telegraf/blob/c6dbc3f689dc3ef22775066cd69b05c60875131d/docs/developers/REVIEWS.md?plain=1#L160-L164) to be consistent with other plugins where we report the server/host/source of the metric.

Reply:
@W0CHP,

I had a look at the values and see:

host: hostname of the instance
sponsor: provider of the instance
id: id of the server
name: location

Isn't host more interesting than sponsor?